index.d.ts

```diff
@@ -0,0 +1,116 @@
+// Type definitions for core-decorators.js 0.10
+// Project: https://github.com/jayphelps/core-decorators.js
+// Definitions by: Qubo <https://github.com/tkqubo>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+export interface ClassDecorator {
+    <TFunction extends Function>(target: TFunction): TFunction | void;
+}
+
+export interface ParameterDecorator {
+    (target: object, propertyKey: string | symbol, parameterIndex: number): void;
+}
+
+export interface PropertyDecorator {
+    (target: object, propertyKey: string | symbol): void;
+}
+
+export interface MethodDecorator {
+    <T>(target: object, propertyKey: string | symbol, descriptor: TypedPropertyDescriptor<T>): TypedPropertyDescriptor<T> | void;
+}
+
+export interface PropertyOrMethodDecorator extends MethodDecorator, PropertyDecorator {
+    (target: object, propertyKey: string | symbol): void;
+}
+
+export interface Deprecate extends MethodDecorator {
+    (message?: string, option?: DeprecateOption): MethodDecorator;
+}
+
+export interface DeprecateOption {
+    url: string;
+}
+
+export interface ThrottleOptions {
+    /** allows to trigger function on the leading. */
+    leading?: boolean;
+    /** allows to trigger function on the trailing edge of the wait interval. */
+    trailing?: boolean;
+}
+
+export interface Console {
+    log(message?: any, ...optionalParams: any[]): void;
+    time(timerName?: string): void;
+    timeEnd(timerName?: string): void;
+}
+
+/**
+ * Forces invocations of this function to always have this refer to the class instance,
+ * even if the function is passed around or would otherwise lose its this context. e.g. var fn = context.method;
+ */
+export const autobind: Function;
+/**
+ * Marks a property or method as not being writable.
+ */
+export const readonly: PropertyOrMethodDecorator;
+/**
+ * Checks that the marked method indeed overrides a function with the same signature somewhere on the prototype chain.
+ */
+export const override: MethodDecorator;
+/**
+ * Calls console.warn() with a deprecation message. Provide a custom message to override the default one. You can also provide an options hash with a url, for further reading.
+ */
+export const deprecate: Deprecate;
+/**
+ * Calls console.warn() with a deprecation message. Provide a custom message to override the default one. You can also provide an options hash with a url, for further reading.
+ */
+export const deprecated: Deprecate;
+/**
+ * Creates a new debounced function which will be invoked after wait milliseconds since the time it was invoked. Default timeout is 300 ms.
+ */
+export function debounce(wait: number): MethodDecorator;
```

Comment:
As `debounce`, `throttle` and `mixin` are [deprecated for this package](https://github.com/jayphelps/core-decorators.js#for-methods), do we really want to include them in the TypeScript declarations?

At a minimum, the JSdoc [@deprecated](http://usejsdoc.org/tags-deprecated.html) annotation should probably be added to their doc comments.

@jayphelps do you have an opinion on this choice?